telemetry/csharp/AwsToolkit.Telemetry.Events/Core/BaseTelemetryEvent.cs

```diff
@@ -23,6 +23,12 @@ public abstract class BaseTelemetryEvent
         /// </summary>
         public string Reason;
 
+        /// <summary>
+        /// Optional - The duration for the workflow associated with the metric 
+        /// This is often used in multi-step workflows to provide additional details about how long did the action take
+        /// </summary>
+        public double Duration;
```

Comment:
Discussed offline: we want to handle this field like an optional (similar to the `Value` property). This means it isn't emitted into json payloads if there isn't a value specified.